#! /bin/sh /usr/share/dpatch/dpatch-run
## 00_0163-fix-gcc43-support.dpatch by Dirk Mueller <mueller@kde.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Various fixes to get Qt 4.3 without hundreds of warnings compiling.

@DPATCH@
diff -urNad qt4-x11-4.3.0~/config.tests/unix/endian.test qt4-x11-4.3.0/config.tests/unix/endian.test
--- qt4-x11-4.3.0~/config.tests/unix/endian.test	2007-05-25 15:24:34.000000000 +0200
+++ qt4-x11-4.3.0/config.tests/unix/endian.test	2007-06-26 19:54:20.000000000 +0200
@@ -30,6 +30,7 @@
     [ "$VERBOSE" = "yes" ] && echo "    Found 'MostSignificantByteFirst' in binary"
     ENDIAN="BIG"
 fi
+[ "$VERBOSE" = "yes" ] && make clean || make clean >/dev/null 2>&1
 
 # done
 if [ "$ENDIAN" = "LITTLE" ]; then
diff -urNad qt4-x11-4.3.0~/config.tests/x11/glib/glib.cpp qt4-x11-4.3.0/config.tests/x11/glib/glib.cpp
--- qt4-x11-4.3.0~/config.tests/x11/glib/glib.cpp	2007-05-25 15:24:25.000000000 +0200
+++ qt4-x11-4.3.0/config.tests/x11/glib/glib.cpp	2007-06-26 19:54:20.000000000 +0200
@@ -6,7 +6,7 @@
 {
     GMainContext *context;
     GSource *source;
-    GPollFD *pollfd;
+    GPollFD *pollfd = NULL;
     if (!g_thread_supported())
         g_thread_init(NULL);
     context = g_main_context_default();
diff -urNad qt4-x11-4.3.0~/src/corelib/codecs/qtextcodec.cpp qt4-x11-4.3.0/src/corelib/codecs/qtextcodec.cpp
--- qt4-x11-4.3.0~/src/corelib/codecs/qtextcodec.cpp	2007-05-25 15:24:09.000000000 +0200
+++ qt4-x11-4.3.0/src/corelib/codecs/qtextcodec.cpp	2007-06-26 19:54:20.000000000 +0200
@@ -415,7 +415,7 @@
             localeMapper = checkForCodec(lang);
 
         // 5. "@euro"
-        if (!localeMapper && ctype && strstr(ctype, "@euro") || lang && strstr(lang, "@euro"))
+        if ((!localeMapper && ctype && strstr(ctype, "@euro")) || (lang && strstr(lang, "@euro")))
             localeMapper = checkForCodec("ISO 8859-15");
 
         // 6. guess locale from ctype unless ctype is "C"
diff -urNad qt4-x11-4.3.0~/src/corelib/io/qtextstream.cpp qt4-x11-4.3.0/src/corelib/io/qtextstream.cpp
--- qt4-x11-4.3.0~/src/corelib/io/qtextstream.cpp	2007-05-25 15:24:10.000000000 +0200
+++ qt4-x11-4.3.0/src/corelib/io/qtextstream.cpp	2007-06-26 19:54:20.000000000 +0200
@@ -541,8 +541,8 @@
     if (!codec || autoDetectUnicode) {
         autoDetectUnicode = false;
         
-        if (bytesRead >= 2 && (uchar(buf[0]) == 0xff && uchar(buf[1]) == 0xfe
-                               || uchar(buf[0]) == 0xfe && uchar(buf[1]) == 0xff)) {
+        if (bytesRead >= 2 && ((uchar(buf[0]) == 0xff && uchar(buf[1]) == 0xfe)
+                               || (uchar(buf[0]) == 0xfe && uchar(buf[1]) == 0xff))) {
             codec = QTextCodec::codecForName("UTF-16");
         } else if (!codec) {
             codec = QTextCodec::codecForLocale();
diff -urNad qt4-x11-4.3.0~/src/corelib/io/qurl.cpp qt4-x11-4.3.0/src/corelib/io/qurl.cpp
--- qt4-x11-4.3.0~/src/corelib/io/qurl.cpp	2007-05-25 15:24:10.000000000 +0200
+++ qt4-x11-4.3.0/src/corelib/io/qurl.cpp	2007-06-26 19:54:21.000000000 +0200
@@ -4724,9 +4724,9 @@
     if (include.isEmpty()) {
         for (int i = 0; i < len; ++i) {
             unsigned char c = *inputData++;
-            if (c >= 0x61 && c <= 0x7A // ALPHA
-                || c >= 0x41 && c <= 0x5A // ALPHA
-                || c >= 0x30 && c <= 0x39 // DIGIT
+            if ((c >= 0x61 && c <= 0x7A) // ALPHA
+                || (c >= 0x41 && c <= 0x5A) // ALPHA
+                || (c >= 0x30 && c <= 0x39) // DIGIT
                 || c == 0x2D // -
                 || c == 0x2E // .
                 || c == 0x5F // _
@@ -4743,9 +4743,9 @@
         const char * alsoEncode = include.constData();
         for (int i = 0; i < len; ++i) {
             unsigned char c = *inputData++;
-            if ((c >= 0x61 && c <= 0x7A // ALPHA
-                || c >= 0x41 && c <= 0x5A // ALPHA
-                || c >= 0x30 && c <= 0x39 // DIGIT
+            if (((c >= 0x61 && c <= 0x7A) // ALPHA
+                || (c >= 0x41 && c <= 0x5A) // ALPHA
+                || (c >= 0x30 && c <= 0x39) // DIGIT
                 || c == 0x2D // -
                 || c == 0x2E // .
                 || c == 0x5F // _
diff -urNad qt4-x11-4.3.0~/src/corelib/tools/qlocale.cpp qt4-x11-4.3.0/src/corelib/tools/qlocale.cpp
--- qt4-x11-4.3.0~/src/corelib/tools/qlocale.cpp	2007-05-25 15:24:11.000000000 +0200
+++ qt4-x11-4.3.0/src/corelib/tools/qlocale.cpp	2007-06-26 19:54:21.000000000 +0200
@@ -2589,8 +2589,8 @@
 
         QChar c1 = format.at(i);
         QChar c2 = format.at(i + 1);
-        if (c1.unicode() == 'a' && c2.unicode() == 'p'
-            || c1.unicode() == 'A' && c2.unicode() == 'P')
+        if ((c1.unicode() == 'a' && c2.unicode() == 'p')
+            || (c1.unicode() == 'A' && c2.unicode() == 'P'))
             return true;
 
         ++i;
diff -urNad qt4-x11-4.3.0~/src/gui/kernel/qapplication.cpp qt4-x11-4.3.0/src/gui/kernel/qapplication.cpp
--- qt4-x11-4.3.0~/src/gui/kernel/qapplication.cpp	2007-05-25 15:24:20.000000000 +0200
+++ qt4-x11-4.3.0/src/gui/kernel/qapplication.cpp	2007-06-26 19:54:21.000000000 +0200
@@ -2018,11 +2018,11 @@
 #ifndef QT_NO_TRANSLATION
 static bool qt_detectRTLLanguage()
 {
-    return force_reverse ^
+    return (force_reverse ^
         QApplication::tr("QT_LAYOUT_DIRECTION",
                          "Translate this string to the string 'LTR' in left-to-right"
                          " languages or to 'RTL' in right-to-left languages (such as Hebrew"
-                         " and Arabic) to get proper widget layout.") == QLatin1String("RTL");
+                         " and Arabic) to get proper widget layout.") == QLatin1String("RTL"));
 }
 #endif
 
diff -urNad qt4-x11-4.3.0~/src/gui/kernel/qkeymapper_x11.cpp qt4-x11-4.3.0/src/gui/kernel/qkeymapper_x11.cpp
--- qt4-x11-4.3.0~/src/gui/kernel/qkeymapper_x11.cpp	2007-05-25 15:24:20.000000000 +0200
+++ qt4-x11-4.3.0/src/gui/kernel/qkeymapper_x11.cpp	2007-06-26 19:54:21.000000000 +0200
@@ -1315,8 +1315,8 @@
         }
 
         if (directionKeyEvent && lastWinId == keyWidget->internalWinId()) {
-            if (keysym == XK_Shift_L && directionKeyEvent == XK_Control_L ||
-                keysym == XK_Control_L && directionKeyEvent == XK_Shift_L) {
+            if ((keysym == XK_Shift_L && directionKeyEvent == XK_Control_L) ||
+                (keysym == XK_Control_L && directionKeyEvent == XK_Shift_L)) {
                 directionKeyEvent = Qt::Key_Direction_L;
             } else if (keysym == XK_Shift_R && directionKeyEvent == XK_Control_R ||
                        keysym == XK_Control_R && directionKeyEvent == XK_Shift_R) {
diff -urNad qt4-x11-4.3.0~/src/gui/kernel/qwidget.cpp qt4-x11-4.3.0/src/gui/kernel/qwidget.cpp
--- qt4-x11-4.3.0~/src/gui/kernel/qwidget.cpp	2007-05-25 15:24:19.000000000 +0200
+++ qt4-x11-4.3.0/src/gui/kernel/qwidget.cpp	2007-06-26 19:54:21.000000000 +0200
@@ -1808,7 +1808,7 @@
 
             for (int i = 0; i < pd->children.size(); ++i) {
                 QWidget *w = qobject_cast<QWidget *>(pd->children.at(i));
-                if (w && !w->isWindow() && !w->testAttribute(Qt::WA_WState_Created))
+                if (w && !w->isWindow() && !w->testAttribute(Qt::WA_WState_Created)) {
                     if (w!=q) {
                         w->create();
                     } else {
@@ -1818,6 +1818,7 @@
                         if (winid)
                             w->raise();
                     }
+                }
             }
         } else {
             q->create();
diff -urNad qt4-x11-4.3.0~/src/gui/painting/qregion_unix.cpp qt4-x11-4.3.0/src/gui/painting/qregion_unix.cpp
--- qt4-x11-4.3.0~/src/gui/painting/qregion_unix.cpp	2007-05-25 15:24:13.000000000 +0200
+++ qt4-x11-4.3.0/src/gui/painting/qregion_unix.cpp	2007-06-26 19:54:21.000000000 +0200
@@ -1929,7 +1929,7 @@
         else
             --isInside;
 
-        if (!inside && !isInside || inside && isInside) {
+        if ((!inside && !isInside) || (inside && isInside)) {
             pWETE->nextWETE = AET;
             pWETE = AET;
             inside = !inside;
diff -urNad qt4-x11-4.3.0~/src/tools/uic/driver.cpp qt4-x11-4.3.0/src/tools/uic/driver.cpp
--- qt4-x11-4.3.0~/src/tools/uic/driver.cpp	2007-05-25 15:24:02.000000000 +0200
+++ qt4-x11-4.3.0/src/tools/uic/driver.cpp	2007-06-26 19:54:21.000000000 +0200
@@ -160,7 +160,7 @@
 
 static bool isAnsiCCharacter(const QChar& c)
 {
-    return c.toUpper() >= QLatin1Char('A') && c.toUpper() <= QLatin1Char('Z')
+    return (c.toUpper() >= QLatin1Char('A') && c.toUpper() <= QLatin1Char('Z'))
            || c.isDigit() || c == QLatin1Char('_');
 }
 
